src/realm/query_engine.hpp

```diff
@@ -2101,25 +2141,76 @@ class OrNode : public ParentNode {
     std::vector<std::unique_ptr<ParentNode>> m_conditions;
 
 private:
+    struct ConditionType {
+        ConditionType(const ParentNode& node)
+            : m_col(node.m_condition_column_key.value)
+            , m_type(typeid(node))
+        {
+        }
+        int64_t m_col;
+        std::type_index m_type;
```

Comment:
I've change to use a `std::type_index` which I think should be more robust than using the hash.